src/Stream/GetProductsStreamer.php

```diff
@@ -0,0 +1,295 @@
+<?php
+
+namespace Findologic\Plentymarkets\Stream;
+
+use Findologic\Plentymarkets\Exception\InternalException;
+use Findologic\Plentymarkets\Product;
+use Findologic\Plentymarkets\Wrapper\WrapperInterface;
+use GuzzleHttp\Psr7\Response;
+use JsonStreamingParser\Listener\RegexListener;
+use JsonStreamingParser\Parser;
+use Log4Php\Logger;
+
+/***
+ * Class GetProductsStreamer
+ * @package Findologic\Plentymarkets
+ */
+class GetProductsStreamer implements StreamerInterface
+{
+    const DIRECTORY = 'stream/getproducts/';
+
+    const METADATA_PAGE = 'page';
+    const METADATA_TOTALS_COUNT = 'totalsCount';
+    const METADATA_IS_LAST_PAGE = 'isLastPage';
+    const METADATA_LAST_PAGE_NUMBER = 'lastPageNumber';
+    const METADATA_FIRST_ON_PAGE = 'firstOnPage';
+    const METADATA_LAST_ON_PAGE = 'lastOnPage';
+    const METADATA_ITEMS_PER_PAGE = 'itemsPerPage';
+
+    /**
+     * @var Logger
+     */
+    protected $log;
+
+    /**
+     * @var WrapperInterface
+     */
+    protected $wrapper;
+
+    /**
+     * GetProductsStreamer constructor.
+     * @param Logger $log
+     * @param WrapperInterface $wrapper
+     */
+    public function __construct(Logger $log, WrapperInterface $wrapper)
+    {
+        $this->log = $log;
+        $this->wrapper = $wrapper;
+    }
+
+    /**
+     * @param Response $response
+     * @return string
+     * @throws InternalException
+     */
+    public function streamToFile(Response $response): string
+    {
+        $fileName = uniqid() . '.json';
+
+        $fileHandle = $this->createFile(self::DIRECTORY, $fileName);
+
+        while (!$response->getBody()->eof()) {
+            fwrite($fileHandle, $response->getBody()->read(1024));
+        }
+
+        fclose($fileHandle);
+
+        return self::DIRECTORY . $fileName;
+    }
+
+    /**
+     * @param resource $destination
+     * @param string $streamedFileName
+     * @return resource
+     */
+    public function streamToFileFromStreamedFile($destination, string $streamedFileName)
```

Comment:
Only used for debugger, shouldn't be in this class. We already have a flag for debugging, just check for this one.